sdk/storage/Azure.Storage.Blobs/src/BlobClient.cs

```diff
@@ -49,1522 +68,334 @@ public class BlobClient
-                    nameof(BlobClient),
-                    message:
-                    $"{nameof(this.Uri)}: {this.Uri}\n" +
-                    $"{nameof(proposedID)}: {proposedID}\n" +
-                    $"{nameof(duration)}: {duration}");
-                try
-                {
-                    return await BlobRestClient.Blob.AcquireLeaseAsync(
-                        this.Pipeline,
-                        this.Uri,
-                        duration: duration,
-                        proposedLeaseId: proposedID,
-                        ifModifiedSince: httpAccessConditions?.IfModifiedSince,
-                        ifUnmodifiedSince: httpAccessConditions?.IfUnmodifiedSince,
-                        ifMatch: httpAccessConditions?.IfMatch,
-                        ifNoneMatch: httpAccessConditions?.IfNoneMatch,
-                        cancellation: cancellation)
-                        .ConfigureAwait(false);
-                }
-                catch (Exception ex)
-                {
-                    this.Pipeline.LogException(ex);
-                    throw;
-                }
-                finally
-                {
-                    this.Pipeline.LogMethodExit(nameof(BlobClient));
-                }
-            }
-        }
-
-        /// <summary>
-        /// The <see cref="RenewLeaseAsync"/> operation renews the blob's
-        /// previously-acquired lease.
-        ///
-        /// The lease can be renewed if the <paramref name="leaseId"/>
-        /// matches that associated with the blob.  Note that the lease may be
-        /// renewed even if it has expired as long as the blob has not been
-        /// modified or leased again since the expiration of that lease.  When
-        /// you renew a lease, the lease duration clock resets.
-        ///
-        /// For more information, see <see href="https://docs.microsoft.com/rest/api/storageservices/lease-blob" />.
-        /// </summary>
-        /// <param name="leaseId">
-        /// The ID of the lease to be renewed.
-        /// </param>
-        /// <param name="accessConditions">
-        /// Optional <see cref="HttpAccessConditions"/> to add
-        /// conditions on renewing a lease.
-        /// </param>
-        /// <param name="cancellation">
-        /// Optional <see cref="CancellationToken"/> to propagate
-        /// notifications that the operation should be cancelled.
+            BlobAccessConditions? blobAccessConditions = default,
+            IProgress<StorageProgress> progressHandler = default,
+            CancellationToken cancellationToken = default) =>
+            this.UploadAsync(
+                content,
+                blobAccessConditions: blobAccessConditions,
+                blobHttpHeaders: blobHttpHeaders,
+                metadata: metadata,
+                progressHandler: progressHandler,
+                cancellationToken: cancellationToken)
+                .EnsureCompleted();
```

Comment:
This calls async method without passing isAsync

Reply:
Good catch - fixed.